### PR TITLE
Add field values quotation.

### LINKED ISSUE #4: Missing quotation of field values
When dumping table data to a file field values can't be quoted. This becomes an issue when a string-like, e.g., `VARCHAR`, value contains chars that are used as a delimiter. When a CSV reader reads such a value it splits it into two or more *false* columns. Therefore a CSV reader encounters an unexpectedly higher count of columns.

### PR DESCRIPTION
**Pull request by ajurcik**
Closes #4.

I did a small code refactor as I moved the `Utils.arrayToString` method with its tests to the `cli` module. I did this because I did not want the `reader` to be dependent on the Apache's `commons-csv` lib which I employed for text serialization.

### CODE REVIEW

File: innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * @author xu.zx
+ * @author Adam Jurcik
+ */
+public class CsvPrinter {
+
+  public static final CSVFormat FIELD_FORMAT_QUOTED;
+
+  static {
+    // delimiter is not used
+    FIELD_FORMAT_QUOTED = CSVFormat.newFormat(',')
```
Comment (neoremind) [inline innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java:34]: By default, what you set makes sense. But in mysql dump file, `null` is treated as `NULL`, would it be better to make that an option?
Also, for `QuoteMode.ALL_NON_NULL`, could you make it as an option in CLI?
I suggest resolve input to enum by the following method call:
```
// org.apache.commons.lang3.EnumUtils
QuoteMode quoteMode = EnumUtils.getEnum(QuoteMode.class, "NON_NUMERIC");
```

File: innodb-java-reader-cli/src/test/java/com/alibaba/innodb/java/reader/CsvPrinterTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (neoremind) [inline innodb-java-reader-cli/src/test/java/com/alibaba/innodb/java/reader/CsvPrinterTest.java:1]: Would you remove the header? Because currently we do not have license header in `src/test/java`. Probably I will add them all in the future.

Comment (ajurcik) [reply innodb-java-reader-cli/src/test/java/com/alibaba/innodb/java/reader/CsvPrinterTest.java]: No problem.

File: innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java (hunk 1 above)
Comment (ajurcik) [reply innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java]: Well, this started to be kind of a pain as the project did not support proper CSV printing before. For example:
* the project's code printed lower case `null`s for null values before,
* the `commons-csv` lib needs an escaping character when you choose not to escape the values, which is the project's default way.

I implemented the CLI options as you suggested. However, the escape character option may be missing now. Furthermore, the project supports delimiters as strings, i.e. not only characters, so that the `common-csv` lib cannot be used to properly quote such delimiters within field values.

To summarize, I am not happy about the additional changes. Instead, I would suggest to either (1) use a standard Linux tool such as `sed` to handle the issue with `null` vs. `NULL` or (2) cover the CSV printing in a much more complete way in the project. However, the (2) exceeds a lot the purpose for which I made the change, sorry.

Sorry for not updating the README. I am ready to do it.

**Comment by ajurcik** [comment]
The `master` does not contain the 1.0.5 release commit now. Was it caused accidentally?

File: innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java (hunk 1 above)
Comment (neoremind) [reply innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java]: I learned your concern, for the null, I agree it it better to be processed by external tool, for the QuoteMode, it is ok to quote all non-null fields.

**Comment by neoremind** [comment]
@ajurcik I am sorry for the trouble brought to you, I accidentally squash the latest 2 commits into one, and miss 1.0.5 commit, would you fetch upstream/master and rebase them? I saw conflicts in `TableReaderImpl`, but the changes is trivial, you can accept the the later version as shown below.
```
<<<<<< add-field-quotation
    if (ibdFilePath != null) {
      ibdFilePath = ibdFilePath.trim();
    }
=======
    checkArgument(StringUtils.isNotEmpty(ibdFilePath), "ibdFilePath should not be empty");
    checkArgument(tableDef != null, "tableDef should not be empty");
    ibdFilePath = ibdFilePath.trim();
>>>>>>> master
```

File: innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+   * @return array string
+   */
+  public String arrayToString(Object[] a, boolean newLine) {
+    if (a == null) {
+      return "null";
+    }
+    // clean StringBuilder
+    row.setLength(0);
+    for (Object value : a) {
+      if (quoteMode != QuoteMode.NONE) {
```
Comment (neoremind) [inline innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java:103]: This is a hot spot method, which may have some downside for performance because of branch prediction miss, we can can see if we need it to be tuned in the future, currently this is OK. Everything else LGTM.

**Comment by neoremind** [comment]
@ajurcik Would you resolve the conflict? Then I can merge this PR.

File: innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java (hunk 3 above)
Comment (ajurcik) [reply innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java:103]: Ok I didn't thought about it in this way, lack of experience probably. How can be this fixed?

Comment (neoremind) [reply innodb-java-reader-cli/src/main/java/com/alibaba/innodb/java/reader/cli/CsvPrinter.java:103]: I think JVM JIT can handle this well, too early optimization is not necessary, so this could be ok.

**Comment by neoremind** [comment]
LGTM
